### PR TITLE
Build improvements

### PR DESCRIPTION
**Pull request by emgarten**
* Update xunit runner to 2.3.1
* Update CLI to 2.0.2 and 1.0.4
* Auto run configure.ps1 instead of erroring out
* Update to nuget.exe 4.4.1

### CODE REVIEW

File: src/NuGet.Core/NuGet.CommandLine.XPlat/NuGet.CommandLine.XPlat.csproj [hunk 1]
```diff
@@ -22,7 +22,7 @@
     <PackageReference Include="System.Runtime.InteropServices.RuntimeInformation" Version="4.0.0" />
     <PackageReference Include="Microsoft.Extensions.CommandLineUtils" Version="1.0.1" />
     <PackageReference Include="System.Runtime.Serialization.Primitives" Version="4.1.1" />
-    <PackageReference Include="Microsoft.Build.Runtime" Version="15.1.1012" />
+    <PackageReference Include="Microsoft.Build.Runtime" Version="15.3.409" />
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/NuGet.CommandLine.XPlat.csproj:25]: This was needed to fix a conflict in dotnet add tests with the cli

File: scripts/funcTests/runFuncTests.sh [hunk 2]
```diff
@@ -18,11 +18,12 @@ pushd $DIR
 # Download the CLI install script to cli
 echo "Installing dotnet CLI"
 mkdir -p cli
-curl -o cli/dotnet-install.sh https://raw.githubusercontent.com/dotnet/cli/d2bbe1faa294012cec60b640e6522e0674224d3f/scripts/obtain/dotnet-install.sh
+curl -o cli/dotnet-install.sh https://raw.githubusercontent.com/dotnet/cli/4bd9bb92cc3636421cd01baedbd8ef3e41aa1e22/scripts/obtain/dotnet-install.sh
 
 # Run install.sh
 chmod +x cli/dotnet-install.sh
-cli/dotnet-install.sh -i cli -c preview -v 1.0.1
+# cli/dotnet-install.sh -i cli -c 2.0 --version 2.0.2
```
Comment (emgarten) [inline scripts/funcTests/runFuncTests.sh:25]: This caused problems, but when we move to netstandard2.0 we will need so this so I'm leaving it commented out for now

Comment (rohit21agrawal) [inline scripts/funcTests/runFuncTests.sh:25]: remove this?

File: build/common.ps1 [hunk 3]
```diff
@@ -220,7 +213,7 @@ Function Install-DotnetCLI {
 
         Invoke-WebRequest $cli.DotNetInstallUrl -OutFile $DotNetInstall
 
-        & $DotNetInstall -Channel 2.0 -i $cli.Root -Architecture $arch
+        & $DotNetInstall -Channel 2.0 -i $cli.Root -Version 2.0.2
```
Comment (emgarten) [inline build/common.ps1:216]: Architecture is determined by the script

File: build/common.ps1 [hunk 4]
```diff
@@ -326,7 +319,9 @@ Function Test-BuildEnvironment {
         [switch]$CI
     )
     if (-not (Test-Path $ConfigureJson)) {
-        Error-Log 'Build environment is not configured. Please run configure.ps1 first.' -Fatal
+        # Run the configure script if it hasn't been executed
+        $configureScriptPath = Join-Path $NuGetClientRoot configure.ps1
+        Invoke-Expression $configureScriptPath
```
Comment (rohit21agrawal) [inline build/common.ps1:324]: can you invoke with -Force and -CI if it's running on CI ?

**Review by rohit21agrawal — APPROVED** [review]
minor comment but great improvements! 🎖

File: scripts/funcTests/runFuncTests.sh (hunk 2 above)
Comment (emgarten) [reply scripts/funcTests/runFuncTests.sh:25]: we will need it soon, I can remove it if there is a real issue with it, but installing both of these works, just not with our tests right now

**Review by jainaashish — APPROVED** [review]
👏

**Comment by emgarten** [comment]
Merged

**Pull request closed without merging**